### PR TITLE
HIVE-14514: Cloning writerOptions when creating delete event writers …

### PR DESCRIPTION
**Pull request by difin**
…in OrcRecordUpdater.

### What changes were proposed in this pull request?

When split-update is enabled for ACID, OrcRecordUpdater creates two sets of writers: one for the insert deltas and one for the delete deltas. The deleteEventWriter is initialized with similar writerOptions as the normal writer, except that it has a different callback handler. Previously, when creating deleteWriterOptions it was needed to instantiate a new writer options similar to regular writer due to lack of clone method in WriterOptions class in Orc code. Now Orc WriterOptions class implements Cloneable and deleteWriterOptions can be instantiated by cloning the regular writer options.

### Why are the changes needed?

Improving code quality.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Unit tests and integration tests.

### CODE REVIEW

**Review by ayushtkn — APPROVED** [review]
LGTM

**Comment by difin** [comment]
Hey @ayushtkn thanks for reviewing! Can you please merge it? I have no write access yet..

**Comment by ayushtkn** [comment]
AFAIK, we have to wait 24 hrs after the first approval, so that people in all timezone can have a look.
Will merge it tomorrow

**Pull request merged**
